lib/fileset/default.nix

```diff
@@ -170,7 +171,7 @@ If a directory does not recursively contain any file, it is omitted from the sto
         lib.fileset.toSource: `root` (${toString root}) is a file, but it should be a directory instead. Potential solutions:
             - If you want to import the file into the store _without_ a containing directory, use string interpolation or `builtins.path` instead of this function.
             - If you want to import the file into the store _with_ a containing directory, set `root` to the containing directory, such as ${toString (dirOf root)}, and set `fileset` to the file path.''
-    else if ! hasPrefix root fileset._internalBase then
+    else if ! fileset._internalIsEmptySingleton && ! hasPrefix root fileset._internalBase then
```

Comment:
This will produce an awkward error message. Maybe check this separately?

Reply:
I don't see how this would produce an awkward error message?